src/ui/components/UploadScreen/SettingsPreview.tsx

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import { Workspace } from "ui/types";
+
+const getIconAndText = (
+  isPublic: boolean,
+  selectedWorkspaceId: string | null,
+  workspaceName: string
+) => {
+  let text, icon;
+
+  if (!isPublic) {
+    if (selectedWorkspaceId) {
+      text = `Shared privately with ${workspaceName}`;
+      icon = "groups";
+    } else {
+      text = `Only you can view this`;
+      icon = "person";
+    }
+  } else {
+    text = "This replay can be viewed by anyone with the link";
+    icon = "public";
```

Comment:
What do you think about introducing a type like `PrivacyMode`.
```
interface PrivacyMode {
  text: string,
  icon: string,
}

const PUBLIC = {
  text: "This replay can be viewed by anyone with the link"
  icon: "public"
}

const ONLY_YOU = {
  ...
}

const privacyMode(isPublic: boolean, workSpaceId: string | null, workspaceName: string): PrivacyMode => {
  if (isPublic) {
    return PUBLIC
  } else if (workSpaceId) {
    return {
    text: `Shared privately with ${workspaceName}`
    icon: "groups",
    }
  } 
​return PRIVATE
}
```

GitHub is hard to write code on, but hopefully you get what I mean?

Reply:
Added the typing just to make sure that we avoid situations where the functions ends up returning something we don't expect. Apart from that, not really a fan of extracting the return values into constants since it's not being reused.